### PR TITLE
Trick get a new copy of object/array using spread operator

### PR DESCRIPTION
**Pull request by wonderingabout**
A common trick using the spread operator :)
Credit goes to @Dorus for teaching me this trick !

### CODE REVIEW

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md [hunk 1]
```diff
@@ -225,6 +225,49 @@ But there's a subtle difference between `Array.from(obj)` and `[...obj]`:
...
+                          // then return the result in a new object
+
+// do the objects have the exact same contents ?
+alert(JSON.stringify(obj) === JSON.stringify(objCopy)); // true (same content)
+
+// are the arrays equal (they share the same reference) ?
+alert(obj === objCopy); // false (not same reference)
+```
+
+This way of copying an object is much shorter than
```
Comment (leviding) [inline 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: Line 267 and 268 should be in one line?

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md [hunk 2]
```diff
@@ -119,7 +119,7 @@ f(1); // 1
 ```
 
 As we remember, arrow functions don't have their own `this`. Now we know they don't have the special `arguments` object either.
-````
+```
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: indeed, fixed

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md [hunk 3]
```diff
@@ -225,6 +225,49 @@ But there's a subtle difference between `Array.from(obj)` and `[...obj]`:
 So, for the task of turning something into an array, `Array.from` tends to be more universal.
 
 
+## Get a new copy of an object/array
+
+Remember when we talked about `Object.assign()`
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: generally, github formatting does automatically a word wrap for one skipped line, the purpose was to keep easy to read code in the github editor
the javascript.info may not behave the same way, so i fixed it to one long line

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md [hunk 4]
```diff
@@ -225,6 +225,49 @@ But there's a subtle difference between `Array.from(obj)` and `[...obj]`:
 So, for the task of turning something into an array, `Array.from` tends to be more universal.
 
 
+## Get a new copy of an object/array
+
+Remember when we talked about `Object.assign()` 
+[in the past](https://javascript.info/symbol#symbols-are-skipped-by-for-in) ?
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: fixed, thanks

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md [hunk 5]
```diff
@@ -225,6 +225,49 @@ But there's a subtle difference between `Array.from(obj)` and `[...obj]`:
...
+
+It is possible to do the exact same thing with the spread operator !
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: fixed, thanks

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md [hunk 6]
```diff
@@ -225,6 +225,49 @@ But there's a subtle difference between `Array.from(obj)` and `[...obj]`:
...
+
+```
+let arr = [1, 2, 3];
+let arrCopy = [...arr]; // spread the array into a list of parameters
+                        // then put the result into a new array
+
+// do the arrays have the exact same contents ?
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: fixed, thanks

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md [hunk 7]
```diff
@@ -225,6 +225,49 @@ But there's a subtle difference between `Array.from(obj)` and `[...obj]`:
...
+alert(JSON.stringify(arr) === JSON.stringify(arrCopy)); // true (same content)
+
+// are the arrays equal (they share the same reference) ?
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: fixed, thanks

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md [hunk 8]
```diff
@@ -225,6 +225,49 @@ But there's a subtle difference between `Array.from(obj)` and `[...obj]`:
...
+
+```
+let obj = { a: 1, b: 2, c: 3 };
+let objCopy = { ...obj }; // spread the object into a list of parameters
+                          // then return the result in a new object
+
+// do the objects have the exact same contents ?
+alert(JSON.stringify(obj) === JSON.stringify(objCopy)); // true (same content)
+
+// are the arrays equal (they share the same reference) ?
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: fixed, thanks

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md (hunk 1 above)
Comment (wonderingabout) [reply 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: same as https://github.com/javascript-tutorial/en.javascript.info/pull/1823#discussion_r399063603

**Comment by paroche** [comment]
Alexey -- is there a specific change you want me to look at? Everything I see above seems to be marked  "outdated".

In the a54f242 branch linked to in the last section there is some awkward language:

'Note that it is possible to do the exact same to copy of objects: '

could be improved:

'Note that it is possible to do the exact same thing to copy objects:'

or 

'Note that it is possible to do the exact same thing to make a copy of an object:'
(my preference)

Rest of the language is OK.

Anything else?

File: 1-js/06-advanced-functions/02-rest-parameters-spread/article.md [hunk 9]
```diff
@@ -225,6 +225,49 @@ But there's a subtle difference between `Array.from(obj)` and `[...obj]`:
...
+alert(arr === arrCopy); // false (not same reference)
+
+// modify our initial array:
+arr.push(4)
+alert(arr) // 1, 2, 3, 4
+alert(arrCopy) // 1, 2, 3
+```
+
+Note that it is possible to do the exact same to copy
```
[the remark this answers is not in the document]
Comment (wonderingabout) [reply 1-js/06-advanced-functions/02-rest-parameters-spread/article.md]: same as https://github.com/javascript-tutorial/en.javascript.info/pull/1823#discussion_r399064232

**Comment by wonderingabout** [comment]
> 'Note that it is possible to do the exact same thing to make a copy of an object:'
> (my preference)

fixed, it's indeed clearer

**Comment by paroche** [comment]
Fine by me!

**Comment by wonderingabout** [comment]
@paroche did you accidentally hit the close PR instead of the merge PR button (just asking myself :thinking: )

**Comment by paroche** [comment]
You could say that. Anyway, as far as I'm concerned it should be merged (and closed).

**Comment by paroche** [comment]
And it is.

**Comment by wonderingabout** [comment]
yes, you can always @ me if we need to change something

**Pull request merged**